### PR TITLE
Feats: Error Manager

### PR DESCRIPTION
**Pull request by Swiizyy**

### CODE REVIEW

File: src/lib/utils/errorHandling.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+			guildId: error.guildId,
+			userId: error.userId,
+			interaction: error.stack ? error.stack.split('\n')[1].trim() : undefined,
+		});
+		scope.setLevel(severity);
+		Sentry.captureException(error);
+	});
+}
+
+function sendErrorMessageToUser(interaction: ErrorHandlingOptions['interaction'], error: ErrorHandlingOptions['error']): void {
```
Comment (nikolaischunk) [inline src/lib/utils/errorHandling.ts:58]: Maybe also send the Error Embed to a Admin Log channel, that we also see what errors are happening.

File: src/lib/utils/errorHandling.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	const errorMessageEmbed = generateEmbed({
+		title: 'Une erreur est survenue',
+		description: codeBlock('shell', `Commande : ${
```
Comment (nikolaischunk) [inline src/lib/utils/errorHandling.ts:61]: Add a kind of UUID to the message, that we can identify which error is ment when someone reports it.

Comment (Swiizyy) [reply src/lib/utils/errorHandling.ts:61]: Yes, Great idea :)

File: src/lib/utils/errorHandling.ts (hunk 1 above)
Comment (Swiizyy) [reply src/lib/utils/errorHandling.ts:58]: I would rather think of a webhook when you say?

**Comment by nikolaischunk** [comment]
grafik
The logs look super wierd now, can you revert this?

**Comment by nikolaischunk** [comment]
What does an Error message look like when sent via Webhook / to the user? Can you send a screenshot here?

**Comment by nikolaischunk** [comment]
> The webhook is not sent to the user but to the channel
I am aware. But how do the two messages look like?
1. The message sent to the user
2. The Webhook sent to the channel

**Comment by Swiizyy** [comment]
> grafik
> 
> The logs look super wierd now, can you revert this to how it was before?

Can you send me a screen from before?

**Comment by nikolaischunk** [comment]
I mocked the error, when fetch a userid that does not exist. The error is printed to the console, but no webhook gets sent or feeback to the user 
grafik
grafik

**Comment by Swiizyy** [comment]
How did you use the function?

**Comment by nikolaischunk** [comment]
@Swiizyy please provide a screenshot of how it looks now, colorwise?

**Comment by Swiizyy** [comment]
Its default params

**Comment by nikolaischunk** [comment]
grafik

**Comment by Swiizyy** [comment]
@nikolaischunk The files have changed place, it is enough to reinstall the dependencies with `yarn install` to reset the framework.
After this order it works perfectly on my side :)

File: src/listeners/container/commands/chatInputCommands/chatInputCommandError.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { ApplyOptions } from '@sapphire/decorators';
+import { ChatInputCommandErrorPayload, Events, Listener } from '@sapphire/framework';
+import { handleCommandErrorAndSendToUser } from '../../../../lib/utils/errorHandling';
+
+@ApplyOptions<Listener.Options>({ event: Events.ChatInputCommandError })
+export class ChatInputCommandErrorEvent extends Listener<typeof Events.ChatInputCommandError> {
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (nikolaischunk) [inline src/listeners/container/commands/chatInputCommands/chatInputCommandError.ts:7]: there is no unused variable?

File: src/listeners/container/commands/contextMenuCommands/contextMenuCommandError.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { ApplyOptions } from '@sapphire/decorators';
+import { ContextMenuCommandErrorPayload, Events, Listener } from '@sapphire/framework';
+import { handleCommandErrorAndSendToUser } from '../../../../lib/utils/errorHandling';
+
+@ApplyOptions<Listener.Options>({ event: Events.ContextMenuCommandError })
+export class ContextMenuCommandErrorEvent extends Listener<typeof Events.ContextMenuCommandError> {
+	// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
Comment (nikolaischunk) [inline src/listeners/container/commands/contextMenuCommands/contextMenuCommandError.ts:7]: there is no unused variable?

**Review by nikolaischunk — CHANGES_REQUESTED** [review]

**Comment by nikolaischunk** [comment]
grafik
Is this log really necessary? I'd suggest to only log the failures not the successful heartbeats.

**Review by nikolaischunk — APPROVED** [review]

**Comment by nikolaischunk** [comment]
Ready to Merge 🎉

**Comment by Swiizyy** [comment]
LGTM :rocket:

**Pull request merged**